### DISCUSSION TITLE
Idea about switching level of tasks below-ground and in upper levels

### DISCUSSION THREAD

**Discussion by moogaem** [streetcomplete/StreetComplete#3458] [Ideas]
Currently StreetCompleet shows all tasks within range. In the case of below-ground structures like subways etc. hence this include tasks on all levels.
When there is a complex topography - like a complex intersection in a city with motorways, bike lanes, tram lines, underground pathways, subway halls etc. it is difficult to find out which tasks are at your current level. You would often click on tasks only to see that they are not on your current level or even answer them wrongly if you don't realize that the task refers to another level.

Therefore I suggest to include a filter where the current level (e.g. 0 for above ground, -1, +1 etc.) can be chosen. There could be a general option to include that filter or to manually activate it. Then the box for chosing a level would pop up as soon as there are tasks on the screen which are on different levels. If you choos a level then only the tasks which are on the current level would be displayed. Maybe a hint like "There are N tasks on other levels." could be displayed.

**Comment by smichel17** [comment]
This has been discussed quite extensively in the past. I'm not sure where all the relevant issues are, but you can start at https://github.com/streetcomplete/StreetComplete/issues/2874

If you search more, please let me know the most useful issues/comments so I can add them to #2789

**Reply by moogaem** [reply]
Thanks for the hint to the previous debate! It seems that the problem has different issues attached wether the POIs on different level are in- or outdoors. And the debate seem to be finished.
Nevertheless when I remember myself stopped mapping at a complex interesection in Munic I still think there is some use for a simple level filter.

**Comment by mnalis** [comment]
Do note that there is a difference between [level=*](https://wiki.openstreetmap.org/wiki/Key:level) and [layer=*](https://wiki.openstreetmap.org/wiki/Key:layer). 

- `level=*` is **exact** level (0=ground etc), but is it used for POIs inside buildings (where it would also be useful to map at which level some POI is, so https://github.com/streetcomplete/StreetComplete/issues/1487).
- for relative height of intersecting highways, bridges etc., however `layer=*` is used, which is only **relative**. So `layer=-7` could be ground level, or `layer=3` could be ground level. No way to know.

Also related: https://github.com/streetcomplete/StreetComplete/issues/3068, https://github.com/streetcomplete/StreetComplete/issues/2874

**Reply by moogaem** [reply]
Thanks for explaining that! So for 'level' would be the adequate variable for filtering POIs.
